File: android/app/src/main/java/com/potatoproject/fries/EngineCache.java
package com.potatoproject.fries;

import android.content.Context;

import io.flutter.embedding.engine.FlutterEngine;
import io.flutter.embedding.engine.FlutterEngineCache;
import io.flutter.embedding.engine.dart.DartExecutor.DartEntrypoint;

public class EngineCache {
    public static FlutterEngine getEngine(Context context) {
        if(!FlutterEngineCache.getInstance().contains("mainEngine")) {
            FlutterEngine engine = new FlutterEngine(context);

            engine
                .getDartExecutor()
                .executeDartEntrypoint(
                    DartEntrypoint.createDefault()
                );
            
            FlutterEngineCache
                .getInstance()
                .put("mainEngine", engine);
        }

        return FlutterEngineCache.getInstance().get("mainEngine");
    }
}

File: android/app/src/main/kotlin/com/potatoproject/fries/EngineCache.kt
package com.potatoproject.fries

import android.content.Context
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.embedding.engine.FlutterEngineCache
import io.flutter.embedding.engine.dart.DartExecutor.DartEntrypoint

class EngineCache {
    companion object {
        fun getEngine(context: Context?): FlutterEngine? {
            if (!FlutterEngineCache.getInstance().contains("mainEngine")) {
                val engine = FlutterEngine(context!!)
                engine.dartExecutor.executeDartEntrypoint(DartEntrypoint.createDefault())
                FlutterEngineCache.getInstance().put("mainEngine", engine)
            }
            return FlutterEngineCache.getInstance()["mainEngine"]
        }
    }
}